### PR TITLE
App launcher displays the funktion "Einchecken", although the access to camera is denied by user

### PR DESCRIPTION
**Pull request by freshking**
## Description

Refactored quick actions code to construct shortcut items when app moved into an inactive state. This way more areas in the code can manipulate the existence of shortcut items outside of the exposure submission flow.

## Link to Jira

https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6237

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate+QuickActions.swift [hunk 1]
```diff
@@ -3,13 +3,62 @@
...
+		// No shortcuts if not onboarded
+		guard let appDelegate = application.delegate as? AppDelegate, appDelegate.store.isOnboarded else {
+			application.shortcutItems = nil
+			return
+		}
+
+		var shortcutItems = [UIApplicationShortcutItem(type: QuickAction.diaryNewEntry.rawValue, localizedTitle: AppStrings.QuickActions.contactDiaryNewEntry, localizedSubtitle: nil, icon: UIApplicationShortcutIcon(templateImageName: "book.closed"))]
+		
+		let status = AVCaptureDevice.authorizationStatus(for: .video)
+		if status == .authorized || status == .notDetermined {
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate+QuickActions.swift:53]: I like the Idee, but is it synced with POs?

Comment (freshking) [reply src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate+QuickActions.swift:53]: I am not sure but it is in sync with the ticket description.

**Review by marcussc — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate+QuickActions.swift [hunk 2]
```diff
@@ -3,13 +3,62 @@
...
 	case diaryNewEntry = "de.rki.coronawarnapp.shortcut.diarynewentry"
 	case eventCheckin = "de.rki.coronawarnapp.shortcut.eventcheckin"
+	
+	static var exposureSubmissionFlowTestResult: TestResult?
+	
+	private static var willResignActiveNotification: NSObjectProtocol?
+	
+	static func setup() {
+		guard willResignActiveNotification == nil else {
+			return
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate+QuickActions.swift:22]: perhaps add a log log here?

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate+QuickActions.swift [hunk 3]
```diff
@@ -3,13 +3,62 @@
...
+	/// Adds or removes quick actions accoring to the current application state
+	private static func update() {
+		
+		Log.info(#function, log: .ui)
+		
+		let application = UIApplication.shared
+				
+		// No shortcuts if test result positiv
+		if let testResult = exposureSubmissionFlowTestResult, testResult == .positive {
+			application.shortcutItems = nil
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate+QuickActions.swift:40]: perhaps add a log log here?

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate+QuickActions.swift [hunk 4]
```diff
@@ -3,13 +3,62 @@
...
+			return
+		}
+
+		// No shortcuts if not onboarded
+		guard let appDelegate = application.delegate as? AppDelegate, appDelegate.store.isOnboarded else {
+			application.shortcutItems = nil
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate+QuickActions.swift:46]: perhaps add a log log here?

**Review by pascal-brause — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate+QuickActions.swift (hunk 3 above)
Comment (freshking) [reply src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate+QuickActions.swift:42]: Added a log

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate+QuickActions.swift (hunk 4 above)
Comment (freshking) [reply src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate+QuickActions.swift:49]: Added a log

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate+QuickActions.swift (hunk 2 above)
Comment (freshking) [reply src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate+QuickActions.swift:22]: Added a log

**Pull request merged**